mitosheet/src/utils/actions.tsx

```diff
@@ -1281,11 +1281,34 @@ export const createActions = (
             searchTerms: ['CodeSnippets'],
             tooltip: "CodeSnippets"
         },
+        [ActionEnum.RESET_AND_KEEP_INDEX]: {
+            type: ActionEnum.RESET_AND_KEEP_INDEX,
+            shortTitle: 'Reset Index',
+            longTitle: 'Reset Index',
+            actionFunction: () => {
+                void mitoAPI.editResetIndex(sheetIndex, false);
+            },
+            isDisabled: () => {return doesAnySheetExist(sheetDataArray) ? undefined : 'Import data before resetting an index.'},
+            searchTerms: ['reset', 'index'],
+            tooltip: "Resets a dataframes index to 0-NUM_ROWS. Will keep the current index as a column in the dataframe."
```

Comment:
0-NUM_ROWS was confusing to me. Maybe we could do `0,1,2...` 

Update in the next `ActionEnum.RESET_AND_DROP_INDEX` action too